### DISCUSSION TITLE
Git installed in two place

### DISCUSSION THREAD

**Discussion by pzrsa** [Homebrew/discussions#430] [Getting started]
hey everyone, I'm new to this whole coding thing especially in web development so please bear with me if I say anything that isn't really asked much. Basically, I installed homebrew the default way with the code on my new MacBook Pro M1 chip with Apple silicon and the warning earlier came up about not installing in the opt path.

`Already up-to-date.
Warning: /opt/homebrew/bin is not in your PATH.
==> Installation successful!` 

However, after many hours of just uninstalling and researching (I think) I fixed the issue so it was all good. Now reinstalling all of the homebrew stuff through ARM style via YouTube video I had to sudo a mkdir for opt so I guess it wasn't serious there. But my problem is that I have no two places for the Git installed. The main place is correct and everything works as it should but will having having it in two spots bring any problems in the future? This is what happens:
`% which -a git
/opt/homebrew/bin/git
/usr/bin/git`

I get this sense of anxiety now knowing that somewhere on my before clean laptop I know have some junk where I can't even delete or anything. I don't think im supposed to delete any usr files anyways so I went go that far. Still a bit annoying as I am quite picky with what I allow on my computer and was just following the Odin Project foundation steps.
I hope I did not offend anyone or anything, I am new to all of this stuff and getting back into more of it as I learned Python 1-2 years ago in my school for exams. I look forward to your responses.

**Comment by gromgit** [comment]
> will having [git] in two spots bring any problems in the future?

The only time it _may_ bring any problems is if you accidentally ran one git when you needed to run the other.

For any command that has multiple instances on your system, which one gets run is very much under your control:
* Fully-qualified invocations only run the specific binary in question, i.e. typing `/usr/bin/git` will always run `/usr/bin/git`
* Unqualified invocations will invoke the binary that comes first in `PATH`, e.g. if `/opt/homebrew/bin` comes before `/usr/bin`, then a simple `git` will always run `/opt/homebrew/bin/git`

These rules apply universally, on both your Terminal sessions and any scripts/programs that you may run.

**Reply by pzrsa** [reply]
Oh ok, makes sense. Is there a possible way to delete the /usr/bin/git file? I just don't like how its just there and not in use. If not, no worries.

**Reply by carlocab** [reply]
There is, but it involves deactivating SIP, which is really something that you shouldn't be doing. You could also break something else if you delete it.

**Reply by pzrsa** [reply]
Thank you for the heads up, I have been so on and off with deleting as many people in different groups are saying "oh its ok it won't damage your system". Really appreciate the transparency.
